### ISSUE TITLE
ERROR: invalid VBR checksum 0xZZZZZZZZ (expected 0xYYYYYYYY).

### ISSUE THREAD

**Issue by vermaden** [relan/exfat#51]
Hi,

I tried to mount EXFAT formatted drive under FreeBSD, but it failed with this error:
`ERROR: invalid VBR checksum 0xZZZZZZZZ (expected 0xYYYYYYYY).`

The **exfatfsck** gave me same error without any chance to fix it.

The solution was to comment this line below in **mount.c** and recompile EXFAT:

```
1:56|w530|vermaden /usr/ports/obj/usr/ports/sysutils/fusefs-exfat/work/exfat-1.2.4/libexfat % diff -u mount.c mount.c.VERMOD.c 
--- mount.c     2017-02-12 01:56:34.945134727 +0100
+++ mount.c.VERMOD.c    2017-02-12 01:56:27.335136016 +0100
@@ -137,7 +137,7 @@
                {
                        exfat_error("invalid VBR checksum 0x%x (expected 0x%x)",
                                        le32_to_cpu(((const le32_t*) sector)[i]), vbr_checksum);
-                       return false;
+                       // return false;
                }
        return true;
 }
```

Filesystem was mounted correctly and works without any other errors.

Please provide this feature with some **'force'** flag or **'debug'** flag.

Regards,
vermaden

**Comment by relan** [comment]
Glad to see someone updated a FreeBSD port. It's 2 versions behind the upstream, but that's much better than staying in 2013.

Ignoring consistency checks is not a solution. Need to figure out why this checksum mismatch happens.

What is your FreeBSD version and hardware architecture? What kind of device is that with exFAT FS?

**Comment by vermaden** [comment]
Hi,

I have created 'bugs' for maintainers to update the port(s):

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=217061
sysutils/exfat-utils: Please update to latest 1.2.6

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=217060
sysutils/fusefs-exfat: Please update to latest 1.2.6

> What is your FreeBSD version and hardware architecture?
FreeBSD 10.3-RELEASE-p11 on amd64.

> What kind of device is that with exFAT FS?
'Typical' USB 2.5 Disl Drive - WD My Passport - 1 TB Drive.

The FS was created with this command:
# mkexfatfs -n 1TB -s 128KB /dev/da0

Regards,
vermaden

**Comment by relan** [comment]
> I have created 'bugs' for maintainers to update the port(s)

That's a good idea, 1.2.5 and 1.2.6 contain important fixes.

> FreeBSD 10.3-RELEASE-p11 on amd64.

Will try to reproduce this issue, but it'll take some becuase I'll have to setup a FreeBSD box.

**Comment by vermaden** [comment]
I have the same problem on Raspberry Pi 2 on FreeBSD 11.0-RELEASE on armv6 architecture if that helps.

```
# mount.exfat /dev/da0 /mnt
FUSE exfat 1.2.4
ERROR: invalid VBR checksum 0x2f823ee7 (expected 0x5823efc).
```

Regards,
vermaden

**Comment by relan** [comment]
Do you get invalid VBR error when you check FS immediately after creation? I.e. what does this command print:

    mkexfatfs /dev/da0 && exfatfsck /dev/da0

**Comment by vermaden** [comment]
The error I submitted here is the first time exfat worked 'bad' on FreeBSD for me.

Here are commands outputs:
```
# mkexfatfs /dev/da0
mkexfatfs 1.2.4
Creating... done.
Flushing... done.
File system created successfully.

# exfatfsck /dev/da0
exfatfsck 1.2.4
Checking file system on /dev/da0.
File system version           1.0
Sector size                 512 bytes
Cluster size                 32 KB
Volume size                7388 MB
Used space                 1178 KB
Available space            7387 MB
Totally 0 directories and 0 files.
File system checking finished. No errors found.

# mount.exfat /dev/da0 /mnt/tmp
FUSE exfat 1.2.4
# echo $?
0

# df -m /mnt/tmp
Filesystem 1M-blocks Used Avail Capacity  Mounted on
/dev/fuse       7388    1  7387     0%    /mnt/tmp

# mount | grep /mnt/tmp
/dev/fuse on /mnt/tmp (fusefs, local, synchronous)

# umount /mnt/tmp

# exfatfsck /dev/da0
exfatfsck 1.2.4
Checking file system on /dev/da0.
File system version           1.0
Sector size                 512 bytes
Cluster size                 32 KB
Volume size                7388 MB
Used space                 1178 KB
Available space            7387 MB
Totally 0 directories and 0 files.
File system checking finished. No errors found.

```

**Comment by relan** [comment]
> The error I submitted here is the first time exfat worked 'bad' on FreeBSD for me.

Oh, now I see what you are talking about. There is a general bug about exfatfsck being unable to fix errors: #3. So I close this in favor of that one.

BTW, I couldn't reproduce this issue in a FreeBSD VM. Let's consider it an accidental failure for now.

**Comment by vermaden** [comment]
For the thread/issue to be complete, here is the exact error message:

> ERROR: invalid VBR checksum 0x2f823ee7 (expected 0x5823efc).

Regards,
vermaden
